domehunter/gRPC-server/hx2dome.proto_server.py

```diff
@@ -0,0 +1,578 @@
+            # not really doing anything with park yet
+            response = hx2dome_pb2.IsComplete(return_code=0, is_complete=True)
+            return response
+
+    def dapiIsFindHomeComplete(self, request, context):
+        """TheSkyX RPC to request the completetion status of dapiFindHome.
+
+        Parameters
+        ----------
+        request : Empty
+            Incoming rpc request, a message of type 'Empty'.
+        context : Dunno
+            GRPC magic thingy.
+
+        Returns
+        -------
+        IsComplete
+            rpc response, a message of type 'IsComplete' containing a
+            return_code that indicates the success/failure of the rpc request.
+            and a boolean is_complete that indicates if the relevant rpc
+            has been completed.
+
+        """
+        print("      PHONING HOME    ")
+        if self.server_testing:
+            response = hx2dome_pb2.IsComplete(return_code=0, is_complete=True)
+            return response
+        else:
+            is_dome_moving = self.dome.dome_in_motion
+            # if dome is not moving and dome.is_home returns True
+            # lets just consider the command complete
+            # TODO: better method of determine command completion
+            is_complete = not is_dome_moving & self.dome.is_home
+            response = hx2dome_pb2.IsComplete(
+                return_code=0, is_complete=is_complete)
+            return response
+
+    def dapiSync(self, request, context):
+        """TheSkyX rpc to request a calibration of the dome followed by a new
+        dome azimuth position.
+
+        Parameters
+        ----------
+        request : AzEl
+            Incoming rpc request, a message of type 'AzEl' containing the
+            desired Azimuth and Elevation positions for the observatory.
+        context : Dunno
+            GRPC magic thingy.
+
+        Returns
+        -------
+        ReturnCode
+            rpc response, a message of type 'ReturnCode' that indicates the
+            success/failure of the rpc request.
+
+        """
+        print("      IM SINKINGGGGGGGGG    ")
+        if self.server_testing:
+            response = hx2dome_pb2.ReturnCode(return_code=0)
+            return response
+        else:
+            return_code = 0
+            try:
+                # First re calibrate dome encoder
+                self.dome.calibrate_dome_encoder_counts()
+                # after calibrating send dome to requested azimuth
+                self.dome.GotoAz(request.az)
+            except:
+                # TODO: proper error handling
+                return_code = 1
+            response = hx2dome_pb2.ReturnCode(return_code=return_code)
+            print(f'Sending: dapiSync complete,'
+                  f' return code={response.return_code}\n')
+            return response
+
+
+def serve(kwargs_dict):
+    """Set up the RPC server to run for a day or until interrupted.
```

Comment:
Actually this runs forever until interrupted, it just sleeps for a day at a time.